src/ucp/proto/proto_am.inl

```diff
@@ -27,27 +27,24 @@ typedef void (*ucp_req_complete_func_t)(ucp_request_t *req, ucs_status_t status)
 static UCS_F_ALWAYS_INLINE ucs_status_t
 ucp_am_copy_header_to_internal_buf(ucp_request_t *req, ucs_status_t status)
 {
-    unsigned mask       = UCP_AM_SEND_FLAG_COPY_HEADER | 
-                          UCP_AM_SEND_PRIV_FLAG_HEADER_COPIED;
-    unsigned flags      = req->send.msg_proto.am.flags & mask;
     void *user_header;
     
-    if ((req->send.msg_proto.am.header_length == 0) ||
-        (flags != UCP_AM_SEND_FLAG_COPY_HEADER)) {
-        return status;
-    }
+    if ((req->send.msg_proto.am.header_length != 0) &&
```

Comment:
`(req->send.msg_proto.am.header_length != 0)` should be last check for perf reason